Sources/Overture/Combining.swift

```diff
@@ -0,0 +1,32 @@
+public func combining<Root, Value, NewValue>(
+  _ getter: @escaping (Root) -> Value,
+  _ combine: @escaping (Value, Value) -> NewValue
+  )
+  -> (Value, Root)
+  -> NewValue {
+    return { value, root in combine(value, getter(root)) }
+}
+
+public func combining<Root, Value>(
+  _ getter: @escaping (Root) -> Value,
+  _ combine: @escaping (inout Value, Value) -> Void
+  )
+  -> (inout Value, Root)
+  -> Void {
+    return { value, root in combine(&value, getter(root)) }
+}
+
+public func their<Root, Value, NewValue>(
```

Comment:
Generalized `their` away from the `Bool` requirement. Not sure if there's a good use for it—and maybe the name will be wrong—but Haskell generalizes `on`: http://hackage.haskell.org/package/base-4.12.0.0/docs/Data-Function.html#v:on